tests/integration/test_tableclasses_clusteredmetadatavectorcassandratable.py

```diff
@@ -264,3 +261,268 @@ def test_crud_partitioned_ann(self, db_session: Session, db_keyspace: str) -> No
+            vector=[3, 2],
+            body_blob="A.0.0",
+            metadata={"bb": "A.0.0"},
+        )
+
+        # explicit-partition inserts
+        t.put(
+            partition_id=("moot", "document_A"),
+            row_id=("page0", 1),
+            vector=[3, 3],
+            body_blob="A.0.1",
+            metadata={"bb": "A.0.1"},
+        )
+        t.put(
+            partition_id=("moot", "document_B"),
+            row_id=("page2", 100),
+            vector=[3, -1],
+            body_blob="B.2.100",
+            metadata={"bb": "B.2.100"},
+        )
+
+        # get with implied partition
+        imp_row = t.get(row_id=("page1", 10))
+        assert imp_row is not None
+        assert imp_row["body_blob"] == "A.1.10"
+
+        # get with explicit partition
+        exp_row = t.get(partition_id=("moot", "document_B"), row_id=("page2", 100))
+        assert exp_row is not None
+        assert exp_row["body_blob"] == "B.2.100"
+
+        # get_partition, fully unspecified row_id
+        full_get_part_bodies = [row["body_blob"] for row in t.get_partition()]
+        assert full_get_part_bodies == ["A.0.-1", "A.0.0", "A.0.1", "A.1.10"]
+
+        # get_partition, partial row_id
+        partial_get_part_bodies = [
+            row["body_blob"] for row in t.get_partition(row_id=("page0",))
+        ]
+        assert partial_get_part_bodies == ["A.0.-1", "A.0.0", "A.0.1"]
+
+        # get_partition, partial row_id with range
+        range_get_part_bodies = [
+            row["body_blob"]
+            for row in t.get_partition(
+                row_id=("page0", Predicate(PredicateOperator.GT, -1))
+            )
+        ]
+        assert range_get_part_bodies == ["A.0.0", "A.0.1"]
+
+        # get_partition, full specification (effectively one-row get)
+        fullspec_get_part_bodies = [
+            row["body_blob"] for row in t.get_partition(row_id=("page0", 0))
+        ]
+        assert fullspec_get_part_bodies == ["A.0.0"]
+
+        qvector = [1, 0.02]
+        # ANN within a partition (implied)
+        imp_ann_bodies = [row["body_blob"] for row in t.ann_search(vector=qvector, n=3)]
+        assert imp_ann_bodies == ["A.0.-1", "A.1.10", "A.0.0"]
+
+        # ANN within a partition (explicit)
+        exp_ann_bodies = [
+            row["body_blob"]
+            for row in t.ann_search(
+                vector=qvector, n=3, partition_id=("moot", "document_B")
+            )
+        ]
+        assert exp_ann_bodies == ["B.2.100"]
+
+        # ANN within a partition with partial row_id specification
+        # [row['body_blob'] for row in t.ann_search(vector=qvector,n=3,row_id=("page0",))]
+        # "ANN ordering by vector requires each restricted column to be indexed except for fully-specified partition keys"
+
+        # ANN within a partition with range on row_id
+        # [row['body_blob'] for row in t.ann_search(vector=qvector,n=3,row_id=("page0",Predicate(PredicateOperator.GT,-1)))]
+        # same error as above
```

Comment:
remove these comments? or leave for future reference?

Reply:
I'd like to keep them. At the moment they are very close to being "the best docs" (sadly)